src/vm/eventpipeeventsource.h

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __EVENTPIPE_EVENTSOURCE_H__
+#define __EVENTPIPE_EVENTSOURCE_H__
+
+#ifdef FEATURE_PERFTRACING
+
+class EventPipeProvider;
+class EventPipeEvent;
+class EventPipeSession;
+
+class EventPipeEventSource
```

Comment:
EventPipeEventSource is the EventPipe EventSource - basically, it will provide events about EventPipe.  Today, the only event is the ProcessInfo event, but you can see this expanding quickly to include things like:

 - The set of enabled providers with keyword, level, etc.
 - Logging when a provider is registered / unregistered.
 - Diagnostics around the circular buffer (e.g. when a buffer is stolen because we've hit the max size of the circular buffer).

This is the just the first use of this class.